replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/legacy/OldSite.java

```diff
@@ -0,0 +1,66 @@
+package org.codice.ditto.replication.api.impl.legacy;
+
+import java.util.Map;
+import org.codice.ditto.replication.api.data.ReplicationSite;
+import org.codice.ditto.replication.api.impl.data.AbstractPersistable;
+
+/** OldSite is a class that we can use to retrieve 0.2.0 version sites from the persistent store */
+public class OldSite extends AbstractPersistable implements ReplicationSite {
```

Comment:
I had to create this class to access the old sites using the persistent store. It's the same as ReplicationSiteImpl aside from the PERSISTENCE_TYPE.

Reply:
❓ Can't it extend the current site? Or could the `fromMap()` method of the `Site` class be able to handle the old version and the new one?

Reply:
I didn't know how that would work with trying to override a static field but I guess I can. I think the from map needs to stay the same however. Our new site's are always going to have a version and we need to read it while our old ones will always be null. we could do a null check in the Site class but I'd rather code the class we're generally going to be using for this one specific case.

Reply:
Then how about just extending the class and leave the type intact override the fromMap() to fill-in the version field as expected by the base class and then call super.fromMap()? That way, you don't pollute the base class and you don't end up creating a bran new type.